packages/expo-cli/src/commands/upload/utils.ts

```diff
@@ -22,6 +58,40 @@ export async function downloadFile(url: string, dest: string): Promise<string> {
     bar.tick(progress.transferred - transferredSoFar);
     transferredSoFar = progress.transferred;
   });
+}
+
+export async function extractLocalEASArtifactAsync(
+  filePath: string,
+  extractedDest: string
+): Promise<string> {
+  if (!filePath.endsWith('tar.gz')) {
+    // No need to extract, copy, or rename the file.
+    // Leave it in place and return the path.
+    return filePath;
+  }
+  // Special use-case for downloading an EAS tar.gz file and unpackaging it.
+  const dir = dirname(extractedDest);
+  await pipeline(fs.createReadStream(filePath), tar.extract({ cwd: dir }, []));
+  // Move the folder contents matching ipa or apk.
+  return await moveFileOfTypeAsync(dir, '{ipa,apk}', extractedDest);
```

Comment:
We don't need to handle the `apk` case (Android uploads are handled by Submission Service).
```suggestion
  return await moveFileOfTypeAsync(dir, 'ipa', extractedDest);
```